### PR TITLE
Bug 2060509: Incorrect installation of ibmcloud vpc csi driver in IBM…

### PR DESCRIPTION
**Pull request by dobsonj**
… Cloud ROKS 4.10
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2060509
/cc @openshift/storage @jsafrane @jeffnowicki
/hold for manual testing

### CODE REVIEW

**Comment by dobsonj** [comment]
Those failures don't look good... but the errors I see are mostly things like this:
```
level=error msg=The bootstrap machine failed to download the release image
level=info msg=Error: Error initializing source docker://registry.build02.ci.openshift.org/ci-op-6j8tt1g6/release@sha256:d41552f5fbf56e3ba5474bf72ed672d669a2b5207fe677ee0394090baca377be: Error reading manifest sha256:d41552f5fbf56e3ba5474bf72ed672d669a2b5207fe677ee0394090baca377be in registry.build02.ci.openshift.org/ci-op-6j8tt1g6/release: unauthorized: authentication required
```
/retest

File: pkg/operator/csidriveroperator/csioperatorclient/types.go [hunk 1]
```diff
@@ -23,6 +23,10 @@ type CSIOperatorConfig struct {
 	ConditionPrefix string
 	// Platform where the driver should run.
 	Platform configv1.PlatformType
+	// StatusFilter is an optional callback to determine whether the operator
+	// should run based on the InfrastructureStatus. Return true for yes, or
+	// false for no. If StatusFilter is nil, it is assumed to return true.
```
Comment (jsafrane) [inline pkg/operator/csidriveroperator/csioperatorclient/types.go:28]: Can you please rephrase it to mention that `true` means that all the other checks (like `Platform` and `RequireFeatureGate`) are still used and will block the operator?

Comment (dobsonj) [reply pkg/operator/csidriveroperator/csioperatorclient/types.go]: Yes, updated.

**Comment by dobsonj** [comment]
Tested IPI install on IBM Cloud with these changes. Install completed successfully, driver and operator were running, and PVC could be provisioned successfully.
```
[jon@dagobah ibm]$ oc get pods -n openshift-cluster-csi-drivers
NAME                                                 READY   STATUS    RESTARTS      AGE
ibm-vpc-block-csi-controller-fb87b55b-t7l9k          5/5     Running   0             14h
ibm-vpc-block-csi-driver-operator-76d57949b4-rwf6c   1/1     Running   1 (14h ago)   14h
ibm-vpc-block-csi-node-5jksb                         3/3     Running   0             14h
ibm-vpc-block-csi-node-j5pxk                         3/3     Running   0             14h
ibm-vpc-block-csi-node-k4gst                         3/3     Running   0             14h
ibm-vpc-block-csi-node-s5kdk                         3/3     Running   0             14h
ibm-vpc-block-csi-node-sb6mb                         3/3     Running   0             14h
ibm-vpc-block-csi-node-vn6pr                         3/3     Running   0             14h

[jon@dagobah ibm]$ oc apply -f validPVC.yaml 
persistentvolumeclaim/csi-block-pvc-good created
[jon@dagobah ibm]$ oc apply -f validPOD.yaml 
pod/example created

[jon@dagobah ibm]$ oc get pvc
NAME                 STATUS   VOLUME                                     CAPACITY   ACCESS M
ODES   STORAGECLASS                AGE
csi-block-pvc-good   Bound    pvc-b3f6c90f-7b76-4a29-9fb7-af6e6b34aedc   10Gi       RWO     
       ibmc-vpc-block-5iops-tier   54s
[jon@dagobah ibm]$ oc get pod
NAME      READY   STATUS    RESTARTS   AGE
example   1/1     Running   0          42s
```
I'll leave the hold for now, hoping we can get this tested on ROKS before merging the fix.

**Comment by dobsonj** [comment]
e2e-azure-csi is failing due to [bug 2060687](https://bugzilla.redhat.com/show_bug.cgi?id=2060687)
And the others don't appear to have anything to do with these changes. Primarily concerned with any error indicating the driver/operator fails to deploy.
/retest
